### PR TITLE
Use optype CREATE for single auto-id index requests

### PR DESCRIPTION
**Pull request by ywelsch**
Changes auto-id index requests to use optype CREATE. According to our docs (https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-index_.html#_create_document_ids_automatically):
> If you don’t specify a document ID when using POST, the op_type is automatically set to create and the index operation generates a unique ID for the document.

This is unfortunately not true, as we are currently using `index` as default optype. To make these auto-id index requests compatible with the new "create-doc" index privilege (which is based on the optype), the default optype is changed to create, just as it is already documented.

### CODE REVIEW

**Comment by ywelsch** [comment]
@elasticmachine run elasticsearch-ci/1
run elasticsearch-ci/2
both unrelated failures

File: server/src/main/java/org/elasticsearch/rest/action/document/RestIndexAction.java [hunk 1]
```diff
@@ -90,6 +97,26 @@ void validateOpType(String opType) {
...
+
+        @Override
+        public String getName() {
+            return "document_create_action";
+        }
+
+        @Override
+        public RestChannelConsumer prepareRequest(RestRequest request, final NodeClient client) throws IOException {
+            assert request.params().get("id") == null : "non-null id: " + request.params().get("id");
+            if (request.params().get("op_type") == null && clusterService.state().nodes().getMinNodeVersion().onOrAfter(Version.CURRENT)) {
```
Comment (henningandersen) [inline server/src/main/java/org/elasticsearch/rest/action/document/RestIndexAction.java]: The rest-api-spec (`index.json`) says default for `op_type` is `index`, I think we should change that to be `create`.

I wonder if we should have an end-goal of removing `op_type` support here as a breaking change in 8.0?

**Review by henningandersen — APPROVED** [review]
LGTM.

File: server/src/main/java/org/elasticsearch/rest/action/document/RestIndexAction.java (hunk 1 above)
Comment (ywelsch) [reply server/src/main/java/org/elasticsearch/rest/action/document/RestIndexAction.java]: The problem is that this API spec mixes auto-id which non-auto-id. In one case, the default op_type is create, in the other case it is index. I have moved the explanation of the default into the description (as we have for many other parameters).

**Pull request merged**